File: benchmark/str_ben.php
<?php
    $start = microtime(true);
	$command = "/usr/bin/echo hi;";
    for ($i = 0; $i < 100; $i++) {
		$command = $command . "/usr/bin/echo hi;";
    }
	$output = shell_exec($command);
    $duration = (microtime(true) - $start) * 1000;
    echo "Duration in miliseconds: " . $duration;

    $start = microtime(true);
	$file = fopen('/tmpfs/lore', 'r');
	$lines = '';
	$haystack = '';
	if ($file) {
		while (($line = fgets($file)) !== false) {
			$haystack .= $line;
		}
		fclose($file);
	}
    $needle = 'Dignissim';
    $iter = 20_000;
    for ($i = 0; $i < $iter; $i++) {
        strpos($haystack, $needle);
    }
    $needle = 'Duis';
    for ($i = 0; $i < $iter; $i++) {
        strpos($haystack, $needle);
    }
    $duration = (microtime(true) - $start) * 1000;
    echo "Duration in miliseconds: " . $duration;
?>

File: benchmark/str_ben.py
#!/usr/bin/env python3
import time

start = time.time()
haystack = []
with open('/tmpfs/lore', 'r') as file:
    lines = file.readlines()
    print("const haystack = '''")
    for line in lines:
        if line.strip() == '':
            continue
        print(f'{line.strip()}')
        #haystack.append(line.strip())
    print("'''")

for _ in range(0):
    needle = "potenti"
    index = next((i for i, s in enumerate(haystack) if needle in s), -1)
    needle = "paorttitor."
    index = next((i for i, s in enumerate(haystack) if needle in s), -1)
    needle = "pulvinar"
    index = next((i for i, s in enumerate(haystack) if needle in s), -1)
    needle = "paorttitor."
    index = next((i for i, s in enumerate(haystack) if needle in s), -1)

#print(f'Duration MS: {(time.time() - start) * 1000}')
